### PR TITLE
Support for inherited fields and created instances

### PR DESCRIPTION
**Pull request by cr3ativ3**
Added support to populate already created instances
Added support to populate inherited fields from superclasses
Added support to override @Selector by hiding inherited field under same name
@Selector on final fields are ignored thus no more IllegalAccessException in such cases
Added eclipse project files to .gitignore

### CODE REVIEW

File: jspoon/src/main/java/pl/droidsonroids/jspoon/Utils.java [hunk 1]
```diff
@@ -12,9 +14,10 @@
 import pl.droidsonroids.jspoon.exception.ConstructorNotFoundException;
 import pl.droidsonroids.jspoon.exception.ObjectCreationException;
 
-class Utils {
+public class Utils {
```
Comment (burnoo) [inline jspoon/src/main/java/pl/droidsonroids/jspoon/Utils.java]: `Utils` class shouldn't be public as it's visible for all library users

File: .gitignore [hunk 2]
```diff
@@ -5,4 +5,8 @@
 .DS_Store
 /build
 /captures
-.externalNativeBuild
\ No newline at end of file
+.externalNativeBuild
+.project
+.classpath
+*/bin/
+*/.settings
```
Comment (burnoo) [inline .gitignore]: Is this change necessary?

File: jspoon/src/main/java/pl/droidsonroids/jspoon/Utils.java [hunk 3]
```diff
@@ -49,6 +58,23 @@
...
     }
 
+    /**
+     * Returns a list of all fields, including inherited, starting from the upper most superclass.
+     * @param target the target subclass
+     * @return all declared fields
+     */
+    public static List<Field> getAllDeclaredFields(Class<?> target) {
+        List<Field> declaredFields = new ArrayList<>();
+        if (target.getSuperclass() == null || target == Object.class || target.isInterface()){
```
Comment (koral--) [inline jspoon/src/main/java/pl/droidsonroids/jspoon/Utils.java]: Code can be reformatted, there should be a space before curly bracket.

**Comment by koral--** [comment]
The logic LGTM, but please reformat entire code and correct javadocs. There are more places similar to pointed in previous comments.

File: jspoon/src/main/java/pl/droidsonroids/jspoon/HtmlAdapter.java [hunk 4]
```diff
@@ -95,6 +112,25 @@ public T fromInputStream(InputStream inputStream, URL baseUrl) throws IOExceptio
         return fromInputStream(inputStream, null, baseUrl);
     }
 
+    /**
+     * Populates {@code T} instance from the provided {@code inputStream}.
+     * <p>
+     * Does not close the {@code InputStream}.
```
[the remark this answers is not in the document]
Comment (cr3ativ3) [reply jspoon/src/main/java/pl/droidsonroids/jspoon/HtmlAdapter.java]: This one is copied from/based on your other method. Assuming lowercased inputStream referes to the argument name rather than java object.  Also usually objects in javadocs are mentioned with` {@link InputStream}` rather than `{@code inputStream}.`

File: .gitignore (hunk 2 above)
Comment (cr3ativ3) [reply .gitignore]: Eclipse IDE sometimes creates (in my case) folders for in-workspace binaries (/bin) and workspace settings (/.settings) folders per subproject in addition to .classpath and .project files. And gitignore is there to ignore IDE related files/folders.

File: jspoon/src/main/java/pl/droidsonroids/jspoon/Utils.java (hunk 1 above)
Comment (cr3ativ3) [reply jspoon/src/main/java/pl/droidsonroids/jspoon/Utils.java]: Done.

File: jspoon/src/main/java/pl/droidsonroids/jspoon/Utils.java (hunk 3 above)
Comment (cr3ativ3) [reply jspoon/src/main/java/pl/droidsonroids/jspoon/Utils.java]: Done.

**Comment by cr3ativ3** [comment]
Updated pull request accordingly. Please see my reply for some comments.

Just a heads up, I have a couple more enhancements in development if you would like to hold off release of these changes and have everything bundled.

File: jspoon/src/main/java/pl/droidsonroids/jspoon/HtmlAdapter.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (koral--) [reply jspoon/src/main/java/pl/droidsonroids/jspoon/HtmlAdapter.java]: OK, we'll check and unify that.
